### ISSUE TITLE
DOOM II Expansions Don't work

### ISSUE THREAD

**Issue by Emendo12** [doomhack/GBADoom#42]
I can't seem to get Doom II expansions to load up like with the DOOM 1 expansions. The '.c' file maker seems to actually be adding the data, as the file sizes go up, but they dont show up in the game.

**Comment by Kippykip** [comment]
Oh interesting.
Are you editing the \GBADoom\source\doom_iwad.c in text editor as well to point it to the right file?

**Comment by Emendo12** [comment]
Yeah. I tried doom 2 master levels. I named the wad 'master' and used the name also in the doom_iwad.c. same with the other expansion. I get a .gba file that is in fact bigger than the normal doom 2 game but it does not let me select an expansion. 

Also, when trying to play doom 2 with an expansion added, an error screen shows up and the game is crashed. So something *is* happening but it isn't the good outcome.

**Comment by Kippykip** [comment]
Ah master levels. From what I remember it's not an IWAD and are usually PWADS with specific maps only that are combined with the Doom 2 IWAD. (Say loading doom2.wad + map.wad +warp 08) etc.
Loading a new level meant restarting the game everytime too. It's not like tnt.wad or plutonia.wad where everything was packed into a single loadable iwad file.

It's most likely this project can't determine what gamemode it is due to missing files + missing resources.
I admit I haven't played the recent steam re-release from bethesda so it may be different now with an episode select screen(?) maybe(?). But this project is based on the PrBoom + the original code.

I think the only version of master levels that was put in a single .WAD file officially was PSXDoom but that will be incompatible due to major engine differences.

I'd say the best way to do it is to use SLADE as a wad editor, modify doom2.wad itself and turn the master levels into one big file but that could be a project in itself.
But it looks like that's been done before: https://doomwiki.org/wiki/MASTERLEVELS.WAD

If you open slade, remove the levels from Doom2.wad and replace them with the ones in that, it should be compatible?

**Comment by fusermarucs** [comment]
Hi both - I've got TNT and Plutonia running on my gba......
Can't remember exactly how I did it - it's months and months ago - but it certainly works!

**Comment by Emendo12** [comment]
Hey, so after experimenting with PRBoom and PRBoom+ on PC, I've concluded that the failure to load DOOM 2 expansions rests within PRBoom itself. They will not load in PRBoom while No Rest For The Living loads in PRBoom+. However, Master Levels do not work with either.
